ingestion/src/metadata/ingestion/connections/builders.py

```diff
@@ -67,7 +71,12 @@ def create_generic_db_connection(
     )
 
     if hasattr(connection, "supportsQueryComment"):
-        listen(engine, "before_cursor_execute", inject_query_header, retval=True)
+        listen(
+            engine,
+            "before_cursor_execute",
+            partial(inject_query_header_by_conn, connection),
```

Comment:
I needed to pass a specific function for Vertica to send the query header in a way that is indeed stored by the query history tables